src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1037,6 +1037,76 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
             }
         }
 
+        /// <summary>
+        /// Engine supports config with views that do not have keyfields specified in the config for MsSQL.
+        /// This Test validates that support. It creates a custom config with a view and no keyfields specified.
+        /// It checks both Rest and GraphQL queries are tested to return Success.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.MSSQL)]
+        public async Task TestEngineSupportViewsWithoutKeyFieldsInConfigForMsSQL()
+        {
+            Dictionary<GlobalSettingsType, object> settings = new()
+            {
+                { GlobalSettingsType.GraphQL, JsonSerializer.SerializeToElement(new GraphQLGlobalSettings(){}) },
+                { GlobalSettingsType.Rest, JsonSerializer.SerializeToElement(new RestGlobalSettings(){}) }
+            };
+
+            DataSource dataSource = new(DatabaseType.mssql)
+            {
+                ConnectionString = GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL)
+            };
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(globalSettings: settings, dataSource: dataSource);
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+            Entity viewEntity = new(
+                Source: JsonSerializer.SerializeToElement("books_view_all"),
+                Rest: true,
+                GraphQL: true,
+                Permissions: new PermissionSetting[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                Relationships: null,
+                Mappings: null
+            );
+
+            configuration.Entities.Add("books_view_all", viewEntity);
+
+            File.WriteAllText(
+                CUSTOM_CONFIG,
+                JsonSerializer.Serialize(configuration, RuntimeConfig.SerializerOptions));
+
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string query = @"{
+                    books_view_alls {
+                        items{
+                            id
+                            title
+                        }
+                    }
+                }";
+
+                object payload = new { query };
+
+                HttpRequestMessage graphQLRequest = new(HttpMethod.Post, "/graphql")
+                {
+                    Content = JsonContent.Create(payload)
+                };
+
+                HttpResponseMessage graphQLResponse = await client.SendAsync(graphQLRequest);
```

Comment:
For graphQL, the response code doesn't necessarily tell you whether this worked. GraphQL requests can return 200 but can contain GraphQL errors. You'll need to check for a graphql error in the response. There should be GraphQL helper functions which can do this for you.

Reply:
updated at line 1104.